app/models/admin_post.rb

```diff
@@ -11,8 +11,8 @@ class AdminPost < ApplicationRecord
   }, _suffix: :comments
 
   belongs_to :language
-  belongs_to :translated_post, class_name: 'AdminPost'
-  has_many :translations, class_name: 'AdminPost', foreign_key: 'translated_post_id'
+  belongs_to :translated_post, class_name: "AdminPost"
+  has_many :translations, class_name: "AdminPost", foreign_key: "translated_post_id", dependent: :destroy
```

Comment:
Could you add a test for this? I'm sure it works fine, but a test will let get our attention if we ever mistakenly remove `dependent: :destroy`. 😆 

The best thing to do is probably copy the existing `"deletes post and redirects without notice"` example and modify it to include a translation, e.g., `let!(:translation) { create(:admin_post, language: create(:language), translated_post: post) }`. That way, we'll have tests for deleting both posts with translations and without them.

Reply:
done